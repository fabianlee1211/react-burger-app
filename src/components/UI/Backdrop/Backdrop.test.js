import React from 'react';
import { shallow } from 'enzyme';
import Backdrop from './Backdrop';

describe('<Backdrop />', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Backdrop show={true} />);
  });

  it('renders Backdrop correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('renders null when show props is false', () => {
    wrapper.setProps({ show: false });
    expect(wrapper.html()).toBeNull();
  });

  it('can be clicked', () => {
    const clickFn = jest.fn();
    wrapper.setProps({ clicked: clickFn });
    wrapper.find('div.Backdrop').simulate('click');
    expect(clickFn).toHaveBeenCalled();
  });
});
