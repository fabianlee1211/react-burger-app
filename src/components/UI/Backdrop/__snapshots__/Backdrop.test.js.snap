// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Backdrop /> renders Backdrop correctly 1`] = `
<div
  className="Backdrop"
/>
`;
